src/source_frames.rs

```diff
@@ -251,18 +266,30 @@ fn drop_frame_tc_adjustment(sections: TimecodeSections, rate: Framerate) -> Fram
     Ok(-adjustment)
 }
 
-fn parse_feet_and_frames_str(matched: regex::Captures) -> FramesSourceResult {
+fn parse_feet_and_frames_str(
+    matched: regex::Captures,
+    format: Option<FilmFormat>,
+) -> FramesSourceResult {
     // If we got a match, these groups had to be present, so we can unwrap them.
+
     let feet = timecode_parse::convert_tc_int(matched.name("feet").unwrap().as_str(), "feet")?;
-    let mut frames =
+    let frames =
         timecode_parse::convert_tc_int(matched.name("frames").unwrap().as_str(), "frames")?;
 
+    let perfs = matched.name("perf");
+
     // Get whether this value was a negative timecode value.
     let is_negative = matched.name("negative").is_some();
 
-    frames += feet * FRAMES_PER_FOOT;
+    let final_format = match (format, perfs) {
```

Comment:
This is clever! Do you mind adding. a bespoke error here though? Right now the message returned to the user is: 

```rust
"04+00.03 is not a known frame-count timecode format"
```

I think it should read something more like:

```text
"Format set to `FF16mm`, but `04+00.03` contains a perfs field, which is disallowed for this format."
```

I also think we don't need to error on the number of perfs. The same way this lib considers "00:00:400:01" a valid value, having an above-legal number is okay to me. It's just falling back to a format that doesn't align that bumps.

Reply:
I think this could still be a TimecodeParseError::UnknownStrFormat, but if you want to add a new variant specifically for this I am okay with that too.

Reply:
In de4db17ba81

Reply:
Looks great. Thanks!